Fix some failures with Julia nightly, add missing `action_homomorphism(Omega::GSetByElements{FinGenAbGroup, S})` method

fingolfin opened the pull request:
The order of generators of the group u changed. It was computed via

    t = character_table("L2(8)")
    N = lcm(map(conductor, t))
    u, mu = unit_group(quo(ZZ, N)[1])

and `mu(u[1])` and `mu(u[2])` are swapped in nightly vs julia stable.

Progress towards #4882

fingolfin committed:
Fix a failure with Julia nightly

The order of generators of the group u changed. It was computed via

    t = character_table("L2(8)")
    N = lcm(map(conductor, t))
    u, mu = unit_group(quo(ZZ, N)[1])

and `mu(u[1])` and `mu(u[2])` are swapped in nightly vs julia stable.

lgoettgens commented on test/Groups/gsets.jl:
```diff
@@ -577,5 +577,5 @@ end
   @test all(o -> conductor(sum(collect(o))) == 1, orbs)
   o = orbs[findfirst(o -> length(o) == 3, orbs)]
-  @test [order(permutation(o, x)) for x in gens(u)] == [1, 3]
+  @test sort([order(permutation(o, x)) for x in gens(u)]) == [1, 3]
```
```suggestion
  @test issetequal([order(permutation(o, x)) for x in gens(u)], [1, 3])
```
why not use `issetequal`?

fingolfin replied:
&lt;snark>my solution requires is 4 bytes shorter&lt;/snark>

I don't care so much as long as we get the pesky temp failures out of the way, and not enough to re-run CI over this, esp since @ThomasBreuer already suggested to replace this with a more robust change (I am all for that, and since Thomas' added these tests in #3833 I fully support him replacing them with even better ones :-).

In the meantime I'll push a fix here for the next test failure...

ThomasBreuer reviewed (commented):
Concerning this test, how much can we actually expect from the generators of the group returned by `unit_group`?
Logically, it might make more sense to ask for the action of this group that is induced on the orbit, and to check the group order.

fingolfin committed:
Fix alnuth tests in Julia nightly

fingolfin commented:
Now a bunch of tests fail because the output for things like `factor(ZZ(-60))` changed, due to a different ordering of the factors.

Another failure is due to the order of entries in the vector returned by `subfields(F)` changed.

And so on etc. etc.

Perhaps the only solution we can apply here for now is to *not* run the doctests in `nightly`... in fact there used to be a time were we deliberately only run doctests in one specific Julia version, because it was not worth the effort trying to keep up with changes to that across multiple Julia versions. Maybe we simply should return to that model?

ThomasBreuer committed:
support `action_homomorphism` for G-sets of `FinGenAbGroup`

(The original aim was to improve a test,
this way I found that this function was missing.)

ThomasBreuer commented:
> Perhaps the only solution we can apply here for now is to not run the doctests in nightly... in fact there used to be a time were we deliberately only run doctests in one specific Julia version, because it was not worth the effort trying to keep up with changes to that across multiple Julia versions. Maybe we simply should return to that model?

The current problem with changed hashing in Julia affects more than the doctests.
In many cases, the differences in the tests are due to the fact that we assumed more "stability" than we could guarantee. 
Making the tests more stable will help, but perhaps the different hashing affects even the correctness of the code in some places.

For doctests, the situation is a bit different. Here we want to show certain outputs, even if for example the ordering of subobjects is not stable. In this sense, it is reasonable to check the doctest outputs only in one Julia version.

lgoettgens commented:
I think this is already an improvement to the current situation. Can we get this in or is there anything holding this back?

fingolfin commented:
@lgoettgens I was just about to comment the same :-)

fingolfin commented:
Someone needs to approve this then
